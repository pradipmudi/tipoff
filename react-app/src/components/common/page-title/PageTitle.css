.pageTitle{
    color:#333 !important;
}

.app-spinner {
    margin-left:1em
}