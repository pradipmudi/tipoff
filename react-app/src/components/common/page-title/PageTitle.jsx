import React from 'react';
import {Title, Icon} from 'bloomer';
import './PageTitle.css'

export default function  PageTitle  ({pageTitle, loading})  {
    
    document.title = 'Tipoff - '+pageTitle;

    return (<Title isSize={2} className="pageTitle">
        {pageTitle}
        {loading ? (<Icon className="fas fa-spin fa-spinner app-spinner" />) 
        : 
        (<React.Fragment></React.Fragment>)}
        
    </Title>
)};
