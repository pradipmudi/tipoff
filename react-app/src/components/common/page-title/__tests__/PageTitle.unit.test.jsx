import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import PageTitle from '../PageTitle';

describe("PageTitle", () => {
    
    it("matches snapshot", () => {
        const tree = renderer.create(<PageTitle pageTitle="Tips" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should display the title passed into the property", () => {
        const wrapper = shallow(<PageTitle pageTitle="My App"></PageTitle>);
        expect(wrapper.hasClass('pageTitle')).toEqual(true);
        expect(wrapper.render().text()).toEqual('My App');
    });
    
});
