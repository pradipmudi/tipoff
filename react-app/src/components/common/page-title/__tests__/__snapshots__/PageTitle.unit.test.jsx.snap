// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle matches snapshot 1`] = `
<h1
  className="title is-2 pageTitle"
>
  Tips
</h1>
`;
